geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java

```diff
@@ -14,78 +14,172 @@
+import org.apache.geode.internal.cache.ClientServerObserver;
+import org.apache.geode.internal.cache.ClientServerObserverAdapter;
+import org.apache.geode.internal.cache.ClientServerObserverHolder;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.test.awaitility.GeodeAwaitility;
+import org.apache.geode.test.dunit.IgnoredException;
 import org.apache.geode.test.dunit.NetworkUtils;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.rules.CacheRule;
+import org.apache.geode.test.dunit.rules.DistributedRule;
 import org.apache.geode.test.junit.categories.ClientSubscriptionTest;
 
 /**
  * Tests Redundancy Level Functionality
  */
 @Category({ClientSubscriptionTest.class})
-public class RedundancyLevelPart1DUnitTest extends RedundancyLevelTestBase {
-
-  @BeforeClass
-  public static void caseSetUp() throws Exception {
-    disconnectAllFromDS();
+public class RedundancyLevelPart1DUnitTest implements Serializable {
+
+  private static final long TIMEOUT_MILLIS = GeodeAwaitility.getTimeout().getValueInMS();
+  private static final String K1 = "k1";
+  private static final String K2 = "k2";
+  private static final String REGION_NAME = "RedundancyLevelTestBase_region";
+  private static final int CONNECTED_SERVERS = 3;
+  private static final int DEFAULT_SOCKET_READ_TIMEOUT = 3000;
+  private static final int DEFAULT_RETRY_INTERVAL = 10;
+  private static final int DEFAULT_CONNECTED_SERVER_COUNT = 4;
+
+  private static String SERVER0;
+  private static String SERVER1;
+  private static String SERVER2;
+  private static String SERVER3;
+  private static PoolImpl pool = null;
+  private static AtomicBoolean failOverDetectionByCCU = new AtomicBoolean(false);
+  private static ClientServerObserver clientServerObserver = null;
+  private static InternalCache cache;
+  private static String hostname;
+
+  private final transient CountDownLatch latch = new CountDownLatch(1);
+
+  private int port0;
+  private int port1;
+  private int port2;
+  private int port3;
+  private VM vm0;
+  private VM vm1;
+  private VM vm2;
+  private VM vm3;
+
+  @Rule
+  public DistributedRule distributedRule = new DistributedRule();
+
+  @Rule
+  public CacheRule cacheRule = new CacheRule();
+
+  @Before
+  public void setUp() {
+    hostname = NetworkUtils.getServerHostName();
+
+    vm0 = getVM(0);
+    vm1 = getVM(1);
+    vm2 = getVM(2);
+    vm3 = getVM(3);
+
+    IgnoredException.addIgnoredException("java.net.SocketException||java.net.ConnectException");
+
+    port0 = vm0.invoke(this::createServerCache);
+    port1 = vm1.invoke(this::createServerCache);
+    port2 = vm2.invoke(this::createServerCache);
+    port3 = vm3.invoke(this::createServerCache);
+
+    SERVER0 = hostname + port0;
+    SERVER1 = hostname + port1;
+    SERVER2 = hostname + port2;
+    SERVER3 = hostname + port3;
+
+    System.setProperty(GEMFIRE_PREFIX + "bridge.disableShufflingOfEndpoints", "true");
```

Comment:
clearProperty would be more correct:
```
System.clearProperty(GEMFIRE_PREFIX + "bridge.disableShufflingOfEndpoints");
```
But there's even a better way to do this with a JUnit Rule. Delete the clearProperty line and then just add this Rule:
```
@Rule 
public DistributedRestoreSystemProperties restoreSystemProperties = new DistributedRestoreSystemProperties();
```
That will auto-clear any system properties that were set during any test.

Reply:
Done, though not in setUp(). Default value for disableShufflingOfEndpoints is false so I removed the "reset" in tearDown().